Userland/killall.cpp

```diff
@@ -13,40 +13,17 @@ static void print_usage_and_exit()
 
 static int kill_all(const String& process_name, const unsigned signum)
 {
-    CFile file("/proc/all");
-    if (!file.open(CIODevice::ReadOnly)) {
-	fprintf(stderr, "killall failed to open /proc/all\n");
-	return 3;
-    }
-
-    for (;;) {
-	auto line = file.read_line(1024);
-      
-	if (line.is_empty())
-	    break;
-
-	auto chomped = String((const char*)line.pointer(), line.size() - 1, Chomp);
-	auto parts = chomped.split_view(',');
-
-	if (parts.size() < 18)
-	    break;
-
-	bool ok = false;
-	pid_t pid = parts[0].to_uint(ok);
-	String name = parts[11];
-
-	if (!ok) {
-	    fprintf(stderr, "killall failed : couldn't convert %s to a valid pid\n", parts[0].characters());
-	    return 4;
-	}
-      
-	if (name == process_name) {	
-	    int ret = kill(pid, signum);
+    CProcessHelper processHelper;
+    HashMap<pid_t, RetainPtr<CProcessInfo>> processes = processHelper.get_map();
```

Comment:
I would use auto:
auto processes = processHelper.get_map();

The CProcessHelper object can be temporary as well:
auto processes = CProcessHelper().get_map();

Reply:
I didn't create a static `get_map()` method because I didn't want to have to retrieve the usernames again... and I don't know how I would call this method if I keep the first one, any idea ?

Reply:
It doesn't need to be static :)
CProcessHelper().get_map(); // this will create a temporary CProcessHelper object and call get_map() on it
CProcessHelper::get_map(); // this will call the static member function get_map() with no object